SConscript

```diff
@@ -25,7 +25,9 @@ unittestpp_env.Library(
       'third-party/UnitTest++/src/Posix/TimeHelpers.cpp' ])
 
 test_env = env.Clone(tools=[unittestpp, sajson])
-test_env.Program('test', ['tests/test.cpp', 'tests/test_no_stl.cpp'])
+test_env.Program('test', ['tests/test.cpp',
+                          'tests/test_no_stl.cpp',
+                          'tests/test_unsorted.cpp'])
```

Comment:
I'm not 100% confident this is safe - would it be legal for the C++ compiler to merge the two implementations (each defined differently in each header) into one in the final executable, resulting in only one of the paths being tested?

I think it would be a better test to generate a second executable (called test_unsorted perhaps) that defined `SAJSON_UNSORTED_OBJECT_KEYS` in all source files.